cfme/common/provider.py

```diff
@@ -361,20 +365,61 @@ def _fill_rsa_endpoint_dicts(self, provider_attributes, connection_configs):
         Helper method for ``self.create_rest``
         """
         if "rsa_keypair" not in self.endpoints:
-            return
+            if "rsa_keypair" not in self.data.endpoints:
+                return
+            else:
+                rsa_credentials = self.get_credentials_from_config(
+                    self.data.endpoints["rsa_keypair"]["credentials"], "ssh"
+                )
+        else:
+            rsa_credentials = self.endpoints["rsa_keypair"].credentials
 
-        endpoint_rsa = self.endpoints["rsa_keypair"]
         if isinstance(provider_attributes["credentials"], dict):
             provider_attributes["credentials"] = [provider_attributes["credentials"]]
 
-        provider_attributes["credentials"].append({
-            "userid": endpoint_rsa.credentials.principal,
-            "auth_key": endpoint_rsa.credentials.secret,
-            "auth_type": "ssh_keypair",
-        })
+        provider_attributes["credentials"].append(
+            {
+                "userid": rsa_credentials.principal,
+                "password": rsa_credentials.secret,
+                "auth_type": "ssh_keypair",
+            }
+        )
+
+    def _fill_amqp_endpoint_dicts(self, provider_attributes, connection_configs):
+        """Fills dicts with AMQP events endpoint data.
+
+        Helper method for ``self.create_rest``
+        """
+        if "events" not in self.endpoints:
```

Comment:
I think this should be more specific, since there could be multiple types of event endpoints. Ran into this before with container endpoints.

Reply:
Even if there are multiple types of event endpoints, it does a further check to make sure that the event is `amqp`. Shouldn't that be fine?